src/config/unit_test/unit_test_components.rs

```diff
@@ -0,0 +1,219 @@
+use std::sync::Arc;
+
+use futures_util::{
+    future,
+    stream::{self, BoxStream},
+    FutureExt, SinkExt, StreamExt,
+};
+use serde::{Deserialize, Serialize};
+use tokio::sync::{oneshot, Mutex};
+use vector_core::{
+    event::Event,
+    sink::{StreamSink, VectorSink},
+    transform::DataType,
+};
+
+use crate::{
+    conditions::Condition,
+    config::{SinkConfig, SinkContext, SourceConfig, SourceContext},
+    sinks::Healthcheck,
+    sources,
+};
+
+#[derive(Debug, Serialize, Deserialize, Default, Clone)]
+pub struct UnitTestSourceConfig {
+    #[serde(skip)]
+    pub events: Vec<Event>,
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "unit_test")]
+impl SourceConfig for UnitTestSourceConfig {
+    async fn build(&self, cx: SourceContext) -> crate::Result<sources::Source> {
+        let events = self.events.clone().into_iter().map(Ok);
+
+        Ok(Box::pin(async move {
+            let mut out = cx.out;
+            let _shutdown = cx.shutdown;
```

Comment:
I assume this is to extend the lifetime of the shutdown trigger until after the `out.send_all` so that the topology shuts down correctly? It could be worth commenting that since it wasn't immediately clear.